### ISSUE TITLE
Electron js support

### ISSUE THREAD

**Issue by andrewgremlich** [jakearchibald/idb#33]
This may seem a little trivial, but I guess it could help some other person.

I decided to mess around with IDB in electron js.  I started to get some errors in the console saying that .open wasn't a function.  So I looked into the package.json file and I saw that the node specific main.js file would reject any method request because it would have to be in a browser.  So all I did was change the start js file in node seeing as electron is  counted as a node environment.

So I don't know if you would just want to delete the node.js specific main.js so it could be used in electronjs for anyone else without too much hassle, or maybe even a little small note in the readme saying to change the node specific one.

I hope this makes sense.

**Comment by KeithHenry** [comment]
That's been created as part of this pull: https://github.com/jakearchibald/idb/pull/27

It causes node.js to always fail with idb, which is not quite right - Electron, Atom or anything else that provides a browser window could access idb in node, so that check needs to be expanded.

**Comment by noahehall** [comment]
let me know how i can help with this

**Comment by andrewgremlich** [comment]
Couldn't the check be something like what jQuery implements?  For example here are the first few lines of jQuery...

```( function( global, factory ) {

	"use strict";

	if ( typeof module === "object" && typeof module.exports === "object" ) {

		// For CommonJS and CommonJS-like environments where a proper `window`
		// is present, execute the factory and get jQuery.
		// For environments that do not have a `window` with a `document`
		// (such as Node.js), expose a factory as module.exports.
		// This accentuates the need for the creation of a real `window`.
		// e.g. var jQuery = require("jquery")(window);
		// See ticket #14549 for more info.
		module.exports = global.document ?
			factory( global, true ) :
			function( w ) {
				if ( !w.document ) {
					throw new Error( "jQuery requires a window with a document" );
				}
				return factory( w );
			};
	} else {
		factory( global );
	}

// Pass this if window is not defined yet
} )( typeof window !== "undefined" ? window : this, function( window, noGlobal ) {
```

**Comment by jakearchibald** [comment]
I think we can solve this by having the node version do a bit of feature detecting. Will have a think.

**Comment by andrewgremlich** [comment]
@jakearchibald  couldn't you check if there was an existing document?

**Comment by jakearchibald** [comment]
https://github.com/jakearchibald/idb/tree/electron-fix should solve the problem. Can someone check?

**Comment by andrewgremlich** [comment]
It works in the project that I'm using it in!

**Comment by jakearchibald** [comment]
Merged, and published as 2.0.1! Thanks for the report & help fixing this. I'm delighted, as always, that people find my silly codes useful 😄
